test/visibility/private/uses/weirdEnumHiding.chpl

```diff
@@ -21,8 +21,8 @@ module weirdEnumHiding {
   }
 
   module D {
-    public use super.C;
-    public use super.B;
+    public use super.C except checkFoo;
+    public use super.B except checkFoo;
```

Comment:
It took me a while to figure out why this change was necessary, since all `checkFoo`s were used in a qualified fashion.  It seems like this effectively neuters qualified access to symbols in D that might conflict with other things, and that worries me.

Reply:
Right, the way I see it is this. By having a public use, `module D` is saying it wants to re-export `C.checkFoo` and `B.checkFoo` but it can't do that (because of the duplicate symbol between `C` and `B` and because `D` also defines the same thing). And, we want to allow `D.someSymbol` to find something that is re-exported (on the argument that re-exporting is an implementation detail).

However, a better way to resolve the problem is to use non-public use statements in `module D`. With that, it works the way you would like and the qualified access works too. But, I don't know why this test was using `public use` inside of `D` and if that is somehow significant for the case it is checking. Would you like me to change it to regular `use` instead of adding the `except` clauses?

Reply:
> By having a public use, `module D` is saying it wants to re-export `C.checkFoo` and `B.checkFoo` but it can't do that (because of the duplicate symbol between `C` and `B` and because `D` also defines the same thing).

It's saying that it both/either wants to re-export `C.checkFoo` as `C.checkFoo` or `checkFoo`.  With a public use, you could write `D.C.checkFoo` to get at that symbol.  If D didn't define its own `checkFoo`, you'd also be able to write `D.checkFoo` or `checkFoo` (if it didn't also re-export from B).  The distinction matters less right now because all three functions take no arguments - where being able to explicitly choose the one defined by D itself matters is when it's less clear which version should be called, such as when coercion on the argument type is possible.  If a user wanted to explicitly ensure they always used each particular version, regardless of if we changed language level prioritization of coercion, they could use qualified access to do it.  Or they would be able to without this change.

> However, a better way to resolve the problem is to use non-public use statements in `module D`. With that, it works the way you would like and the qualified access works too. But, I don't know why this test was using `public use` inside of `D` and if that is somehow significant for the case it is checking. Would you like me to change it to regular `use` instead of adding the `except` clauses?

The test was originally added to support https://github.com/chapel-lang/chapel/blob/main/test/visibility/private/uses/use_weirdEnumHiding.chpl in #13355 - I was seeing a bug where `private use A -> public use B` was causing us to skip later uses of B in the same scope.  With that in mind, I do think they need to stay `public`.

Additionally, iirc we already have tests to ensure that more limited use statements (`use A only x;`, `use A except bar;`, etc) don't prevent later unlimited uses of the same module (`use A;`), so I think if we don't think this should be possible, it should either be removed as a test or changed to track a failure mode.